nixos/tests/public-inbox.nix

```diff
@@ -189,12 +200,37 @@ import ./make-test-python.nix (
 
     testScript = ''
       start_all()
-      machine.wait_for_unit("multi-user.target")
-      machine.wait_for_unit("public-inbox-init.service")
 
-      # Very basic check that Gitolite can work;
-      # Gitolite is not needed for the rest of this testScript
+      # The threshold and/or hardening may have to be changed with new features/checks
+      with subtest("systemd hardening thresholds"):
+        print(machine.succeed("systemd-analyze security public-inbox-httpd.service --threshold=5 --no-pager"))
+        print(machine.succeed("systemd-analyze security public-inbox-imapd.service --threshold=5 --no-pager"))
+        print(machine.succeed("systemd-analyze security public-inbox-nntpd.service --threshold=4 --no-pager"))
+
+      with subtest("can setup ssh keys on system"):
+          machine.succeed("install -D -m 600 ${ssh-keys.snakeOilPrivateKey} ~root/.ssh/id_ed25519")
+      machine.wait_for_unit("sshd.service")
+      machine.wait_for_open_port(22)
+
       machine.wait_for_unit("gitolite-init.service")
+      with subtest("create gitolite repositories"):
+          machine.succeed(
+              "git config --global user.name 'System Administrator'",
+              "git config --global user.email root@machine.${domain}",
+              "git clone gitolite@machine.${domain}:gitolite-admin.git",
+              "cat >>gitolite-admin/conf/gitolite.conf ${
+                pkgs.writeText "gitolite-admin-conf-snippet" (
+                  lib.concatMapStringsSep "\n" (repositoryName: ''
+                    repo ${repositoryName}
+                      RW+ = @all
+                  '') gitoliteRepositories
+                )
+              }",
+              "(cd gitolite-admin && git add . && git commit -m 'Add repositories' && git push)",
```

Comment:
Why is this being done over SSH on the same machine?

Reply:
Cloning `gitolite-admin.git` and pushing configurations through SSH is the normal way to control Gitolite, that's [why a pubkey is required](https://github.com/NixOS/nixpkgs/blob/309e8a5e50b782930986a23f41bc91186cac38c6/nixos/modules/services/misc/gitolite.nix#L245) (well, unless `repositories` already exists, but I think it's better to not rely on the internals of the module, and IMHO it's simpler to use SSH).